src/bootstrap/bootstrap.py

```diff
@@ -1162,6 +1162,43 @@ def build_triple(self):
         config = self.get_toml("build")
         return config or default_build_triple(self.verbose)
 
+    def is_git_repository(self, repo_path):
+        return os.path.isdir(os.path.join(repo_path, ".git"))
+
+    def get_latest_commit(self, repo_path, author_email):
+        try:
+            if (
+                not repo_path
+                or not author_email
+                or not self.is_git_repository(repo_path)
+            ):
+                return "<commit>"
+            cmd = [
+                "git",
+                "-C",
+                repo_path,
+                "rev-list",
+                "--author",
+                author_email,
+                "-n1",
+                "HEAD",
+            ]
+            commit = subprocess.check_output(cmd, text=True).strip()
+            return commit if commit else "<commit>"
+        except subprocess.CalledProcessError:
+            return "<commit>"
+
+    def get_value(self):
+        file_path = f"{self.rust_root}/src/stage0"
+        target_key = "git_merge_commit_email"
+        if not os.path.exists(file_path):
+            return None
```

Comment:
Why this is needed?

Maybe just use `parse_stage0_file` instead of creating `get_value` function?